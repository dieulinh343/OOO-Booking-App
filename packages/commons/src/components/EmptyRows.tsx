import React from 'react';
import { Skeleton } from '@ahaui/react';

interface EmptyRowsProps {
  numberOfCols: number;
  numberOfRows: number;
}

export const EmptyRows = ({ numberOfCols, numberOfRows }: EmptyRowsProps): JSX.Element => {
  const cols = Array.from(Array(numberOfCols).keys());
  const rows = Array.from(Array(numberOfRows).keys());

  return (
    <>
      {rows.map(row => (
        <tr key={row}>
          {cols.map(col => (
            <td key={col}><Skeleton width="50%" /></td>
          ))}
        </tr>
      ))}
    </>
  );
};

export default React.memo<EmptyRowsProps>(EmptyRows);
