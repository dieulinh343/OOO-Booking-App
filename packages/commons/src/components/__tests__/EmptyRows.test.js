import React from 'react';
import { shallow } from 'enzyme';
import { EmptyRows } from '../EmptyRows';

describe('components/EmptyRows', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    wrapper = shallow(<EmptyRows {...props} />);
  });

  it('should render empty rows successfully', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
