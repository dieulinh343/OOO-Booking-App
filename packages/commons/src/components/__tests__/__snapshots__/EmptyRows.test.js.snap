// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EmptyRows should render empty rows successfully 1`] = `
<Fragment>
  <tr
    key="0"
  >
    <td
      key="0"
    >
      <Skeleton
        duration={2000}
        height={16}
        variant="text"
        width="50%"
      />
    </td>
  </tr>
</Fragment>
`;
